src/test/ui/associated-const/lints-used-unused.rs

```diff
@@ -0,0 +1,27 @@
+// Tests that associated constants are checked whether they are used or not.
```

Comment:
What is the benefit of adding a new test file for this, instead of using the existing one?

In particular, this test is less useful as it fails tot est both debug and release mode.
For the "used" and "unused" thing, just have two constants, one that is used and one that is not.

Reply:
> What is the benefit of adding a new test file for this, instead of using the existing one?

Compilation stops at this first error, otherwise -- is this something that should be addressed though compilation fails anyway?

> In particular, this test is less useful as it fails to test both debug and release mode.
For the "used" and "unused" thing, just have two constants, one that is used and one that is not.

Ah, true. I'll fix that up now.

Reply:
> Compilation stops at this first error, otherwise -- is this something that should be addressed though compilation fails anyway?

No it does not? That test already contains multiple errors. Making it contain more of them should work just fine.

Reply:
There may still be some `abort_if_errors` around, not sure. The lint could be turned into `warn` to work around any issues with that.

Reply:
Yeah, it didn't when I ran stage1 rustc manually on the test case. 

Ended needing `--emit=link` + changing the lints to warn to get all lints.